scheduler/reconcile_util.go

```diff
@@ -505,7 +505,14 @@ func updateByReschedulable(alloc *structs.Allocation, now time.Time, evalID stri
 	var eligible bool
 	switch {
 	case isDisconnecting:
-		rescheduleTime, eligible = alloc.NextRescheduleTimeByTime(now)
+		replace := alloc.ShouldBeReplaced()
+		if replace != nil {
+			eligible = *replace
+			rescheduleTime = now
+		} else {
+			// Kept to maintain backwards compatibility with behavior prior to 1.8.0
+			rescheduleTime, eligible = alloc.NextRescheduleTimeByTime(now)
+		}
```

Comment:
Rather than returning a `*bool` pointer from `ShouldBeReplaced`, I wonder if we could have it call `NextRescheduleTimeByTime` in the nil case instead? That'd avoid having the potential nil pointer we need to check. I also wonder if "ShouldBeReplaced" is a good name for that method... it only applies in the case of a disconnected alloc, right? 

So all that leads me to think we want something like:


```go
func (a *Allocation) NextRescheduleTimeOnDisconnect(now time.Time) (time.Time, bool) {
	tg := a.Job.LookupTaskGroup(a.TaskGroup)
	if tg == nil || tg.Disconnect == nil || tg.Disconnect.Replace == nil {
		return a.NextRescheduleTimeByTime(now)
	}

	return now, *tg.Disconnect.Replace
}
```

Reply:
It looks cleaner for sure